lib/client-api/src/story_store.ts

```diff
@@ -145,15 +145,20 @@ export default class StoryStore extends EventEmitter {
       data === undefined ? this._selection : { storyId: data.storyId, viewMode: data.viewMode };
     this._error = error === undefined ? this._error : error;
 
-    setTimeout(() => {
-      // preferred method to emit event.
+    try {
       if (this._channel) {
         this._channel.emit(Events.STORY_RENDER);
       }
-
-      // should be deprecated in future.
-      this.emit(Events.STORY_RENDER);
-    }, 1);
+    } catch (e) {
+      setTimeout(() => {
+        if (this._channel) {
+          this._channel.emit(Events.STORY_RENDER);
+        } else {
+          // should be deprecated in future.
+          this.emit(Events.STORY_RENDER);
```

Comment:
This was always getting called before but now is rarely if ever going to be called. I don't think that's right.